tests/phpunit/includes/test-wc-gateway-amazon-payments-advanced.php

```diff
@@ -136,4 +123,118 @@ public function test_classic_process_payment_succeeds() : void {
 			$mock_gateway->process_payment( $order->get_id() )
 		);
 	}
+
+	/**
+	 * Test maybe_separator_and_checkout_button_single_product method.
+	 *
+	 * @return void
+	 */
+	public function test_maybe_separator_and_checkout_button_single_product() : void {
+		ob_start();
+		self::$gateway->maybe_separator_and_checkout_button_single_product();
+		$should_fail_because_no_global_product = ob_get_clean();
+
+		$this->assertEmpty( $should_fail_because_no_global_product );
+
+		$test_product = WC_Helper_Product::create_and_optionally_save_simple_product( true );
+
+		$test_product->set_stock_status( 'outofstock' );
+
+		$test_product->save();
+
+		$GLOBALS['post'] = get_post( $test_product->get_id() );
+
+		ob_start();
+		self::$gateway->maybe_separator_and_checkout_button_single_product();
+		$should_fail_because_out_of_stock = ob_get_clean();
+
+		$this->assertEmpty( $should_fail_because_out_of_stock );
+
+		$test_product->set_stock_status( 'instock' );
+
+		$test_product->save();
+
+		$GLOBALS['post'] = get_post( $test_product->get_id() );
+
+		ob_start();
+		self::$gateway->maybe_separator_and_checkout_button_single_product();
+		$should_fail_because_plugin_option_is_not_enabled = ob_get_clean();
+
+		$this->assertEmpty( $should_fail_because_plugin_option_is_not_enabled );
+
+		// Enable gateway and update single button settings.
+		$this->make_gateway_available( array( 'product_button' => 'yes' ) );
+		self::$gateway->init_settings();
+
+		$expected_for_success = self::$gateway->checkout_button( false, 'div', 'pay_with_amazon_product' );
+
+		ob_start();
+		self::$gateway->maybe_separator_and_checkout_button_single_product();
+		$actual_markup = ob_get_clean();
+
+		$this->assertEquals( $expected_for_success, $actual_markup );
+	}
+
+	/**
+	 * Test load_scripts_on_product_pages method.
+	 *
+	 * @return void
+	 */
+	public function test_load_scripts_on_product_pages() {
+		// If given true it should always return true, no matter what.
+		$this->assertTrue( self::$gateway->load_scripts_on_product_pages( true ) );
+
+		// Should return false, since this is not a product page.
+		$this->assertFalse( self::$gateway->load_scripts_on_product_pages( false ) );
+
+		$test_product = WC_Helper_Product::create_and_optionally_save_simple_product( true );
+
+		$test_product->set_stock_status( 'outofstock' );
+
+		$test_product->save();
+
+		// Force the global post to the newly created test product.
+		$GLOBALS['post'] = get_post( $test_product->get_id() );
+
+		// Force the global query to be a singular product query.
+		$GLOBALS['wp_query']->is_singular    = true;
+		$GLOBALS['wp_query']->queried_object = $GLOBALS['post'];
+
+		// Scripts should not be loaded since product is out of stock.
+		$this->assertFalse( self::$gateway->load_scripts_on_product_pages( false ) );
+
+		$test_product->set_stock_status( 'instock' );
```

Comment:
For readability purposes, can we drop the empty lines before and after this line?